Cannot run test under rustc 1.47.0-nightly (bf4342114 2020-08-25)

I'm using rustc nightly version:
```
rustc 1.47.0-nightly (bf4342114 2020-08-25)
```

Clone the repo and run `cargo test`, below error shown:

```
   Compiling version_check v0.9.2
   Compiling cfg-if v0.1.10
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling syn v1.0.48
   Compiling typenum v1.12.0
   Compiling libc v0.2.80
   Compiling memchr v2.3.4
   Compiling cache-padded v1.1.1
   Compiling futures-io v0.3.7
   Compiling futures-core v0.3.7
   Compiling log v0.4.11
   Compiling getrandom v0.1.15
   Compiling fastrand v1.4.0
   Compiling event-listener v2.5.1
   Compiling autocfg v1.0.1
   Compiling pin-project-lite v0.1.11
   Compiling subtle v2.3.0
   Compiling waker-fn v1.1.0
   Compiling parking v2.0.0
   Compiling once_cell v1.4.1
   Compiling serde_derive v1.0.117
   Compiling proc-macro-hack v0.5.19
   Compiling byteorder v1.3.4
   Compiling vec-arena v1.0.0
   Compiling lazy_static v1.4.0
   Compiling async-task v4.0.3
   Compiling serde v1.0.117
   Compiling percent-encoding v2.1.0
   Compiling ppv-lite86 v0.2.9
   Compiling matches v0.1.8
   Compiling opaque-debug v0.2.3
   Compiling const_fn v0.4.2
   Compiling ryu v1.0.5
   Compiling atomic-waker v1.0.0
   Compiling pin-utils v0.1.0
   Compiling tinyvec v0.3.4
   Compiling maybe-uninit v2.0.0
   Compiling slab v0.4.2
   Compiling serde_json v1.0.59
   Compiling itoa v0.4.6
   Compiling cpuid-bool v0.1.2
   Compiling opaque-debug v0.3.0
   Compiling anyhow v1.0.33
   Compiling base64 v0.12.3
   Compiling data-encoding v2.3.0
   Compiling http-types v2.6.0
   Compiling httparse v1.3.4
   Compiling ansi_term v0.11.0
   Compiling stable_deref_trait v1.2.0
   Compiling infer v0.2.3
   Compiling remove_dir_all v0.5.3
   Compiling difference v2.0.0
   Compiling concurrent-queue v1.2.2
   Compiling async-mutex v1.4.0
   Compiling simple-mutex v1.1.5
   Compiling unicode-bidi v0.3.4
   Compiling form_urlencoded v1.0.0
   Compiling generic-array v0.14.4
   Compiling standback v0.2.11
   Compiling time v0.2.22
   Compiling cookie v0.14.2
   Compiling crossbeam-utils v0.7.2
   Compiling pretty_assertions v0.6.1
   Compiling async-dup v1.2.2
   Compiling unicode-normalization v0.1.13
   Compiling async-channel v1.5.1
   Compiling futures-lite v1.11.2
   Compiling kv-log-macro v1.0.7
   Compiling polling v2.0.2
   Compiling nb-connect v1.0.2
   Compiling num_cpus v1.13.0
   Compiling quote v1.0.7
   Compiling rand_core v0.5.1
   Compiling rand_chacha v0.2.2
   Compiling idna v0.2.0
   Compiling crossbeam-queue v0.2.3
   Compiling byte-pool v0.2.2
   Compiling rand v0.7.3
   Compiling async-io v1.1.10
   Compiling async-executor v1.3.0
   Compiling blocking v1.0.2
   Compiling digest v0.9.0
   Compiling block-cipher v0.7.1
   Compiling universal-hash v0.4.0
   Compiling crypto-mac v0.8.0
   Compiling block-buffer v0.9.0
   Compiling aead v0.3.2
   Compiling aes-soft v0.4.0
   Compiling polyval v0.4.1
   Compiling hmac v0.8.1
   Compiling sha2 v0.9.1
   Compiling hkdf v0.9.0
   Compiling ghash v0.3.0
   Compiling async-global-executor v1.4.3
   Compiling aes v0.4.0
   Compiling aes-gcm v0.6.0
   Compiling tempfile v3.1.0
   Compiling time-macros-impl v0.1.1
   Compiling async-attributes v1.1.1
   Compiling thiserror-impl v1.0.21
   Compiling async-std v1.6.5
   Compiling time-macros v0.1.1
   Compiling thiserror v1.0.21
   Compiling duplexify v1.2.2
   Compiling async-test v1.0.0
   Compiling url v2.1.1
   Compiling serde_urlencoded v0.7.0
   Compiling serde_qs v0.7.0
   Compiling async-h1 v2.1.3 (/Users/jm/github/async-h1)
error[E0308]: mismatched types
   --> src/chunked/decoder.rs:527:38
    |
527 |             let sender = Sender::new(s);
    |                                      ^ expected struct `async_channel::Sender`, found struct `async_std::sync::Sender`
    |
    = note: expected struct `async_channel::Sender<http_types::Trailers>`
               found struct `async_std::sync::Sender<_>`

error[E0308]: mismatched types
   --> src/chunked/decoder.rs:553:38
    |
553 |             let sender = Sender::new(s);
    |                                      ^ expected struct `async_channel::Sender`, found struct `async_std::sync::Sender`
    |
    = note: expected struct `async_channel::Sender<http_types::Trailers>`
               found struct `async_std::sync::Sender<_>`

error[E0308]: mismatched types
   --> src/chunked/decoder.rs:583:38
    |
583 |             let sender = Sender::new(s);
    |                                      ^ expected struct `async_channel::Sender`, found struct `async_std::sync::Sender`
    |
    = note: expected struct `async_channel::Sender<http_types::Trailers>`
               found struct `async_std::sync::Sender<_>`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `async-h1`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Any clue how to fix that?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,6 +20,7 @@ byte-pool = "0.2.1"
 lazy_static = "1.4.0"
 futures-core = "0.3.1"
 log = "0.4"
+async-channel = "1.5.1"
 
 [dev-dependencies]
 pretty_assertions = "0.6.1"

--- a/src/chunked/decoder.rs
+++ b/src/chunked/decoder.rs
@@ -523,7 +523,7 @@ mod tests {
                     .as_bytes(),
             );
 
-            let (s, _r) = async_std::sync::channel(1);
+            let (s, _r) = async_channel::bounded(1);
             let sender = Sender::new(s);
             let mut decoder = ChunkedDecoder::new(input, sender);
 
@@ -549,7 +549,7 @@ mod tests {
             input.extend(vec![b'Z'; 2048]);
             input.extend(b"\r\n0\r\n\r\n");
 
-            let (s, _r) = async_std::sync::channel(1);
+            let (s, _r) = async_channel::bounded(1);
             let sender = Sender::new(s);
             let mut decoder = ChunkedDecoder::new(async_std::io::Cursor::new(input), sender);
 
@@ -579,7 +579,7 @@ mod tests {
                  \r\n"
                     .as_bytes(),
             );
-            let (s, r) = async_std::sync::channel(1);
+            let (s, r) = async_channel::bounded(1);
             let sender = Sender::new(s);
             let mut decoder = ChunkedDecoder::new(input, sender);
 

